test/FileList.cmake

```diff
@@ -9,34 +9,235 @@ set(TestSources
 )
 
 set(TestHeaders
-
+    lib/NumLib.h
 )
 
-set(GOMCSources
+set(GOMCSources 
+   src/BlockOutput.cpp
+   src/BondAdjacencyList.cpp
+   src/BoxDimensions.cpp
+   src/BoxDimensionsNonOrth.cpp
+   src/CBMC.cpp
+   src/CellList.cpp
+   src/ConfigSetup.cpp
+   src/ConsoleOutput.cpp
```

Comment:
Why do we need to rearrange these?

Reply:
1. Yes, for BlockAverage 
  if(restartFromCheckpoint || initStepRead) && step == startStep)) then BlockOutput doesn't write.  The same logic applies to EnCntPartSample.

2.  Yes,  when outputting the merged_psf, the following logic is required to write:
printOnFirstStep -- True for PSF; false for all other classes.  
startStep -- default 0; can be overwritten by InitStep.
Also, merged PSF will return if restartFromCheckpoint is true; no matter what the startStep is.
if((printOnFirstStep && step == **startStep**) || existing logic...)

This is just to replace the hardcoded DoOutput(0) from the PSF constructor.  The startStep can be modified by checkpoint and InitStep after the constructor, so we needed to move it out the constructor.

Reply:
Within the merged PSF method,
  if (restartFromCheckpoint || step != startStep) 
          return;

Reply:
> Why do we need to rearrange these?

There not rearranged.  It's the exact same order.  The only difference is Main.cpp has been removed.

Reply:
The GOMC_Tests link to compiled GOMC code, but they directly build and run Simulation objects; also GTest has it's own Main function, so we can compile code with 2 Main functions.